Source/Scene/Cesium3DTileOptimizations.js

```diff
@@ -0,0 +1,125 @@
+/*global define*/
+define([
+        '../Core/Cartesian3',
+        '../Core/defaultValue',
+        '../Core/freezeObject',
+        './TileOrientedBoundingBox',
+        './TileBoundingRegion'
+    ], function(
+        Cartesian3,
+        defaultValue,
+        freezeObject,
+        TileOrientedBoundingBox,
+        TileBoundingRegion) {
+    'use strict';
+
+    /**
+     * @alias Cesium3DTileOptimizationFlag
+     *
+     * Flag denoting support for a 3D Tiles optimization
+     */
+    var Cesium3DTileOptimizationFlag = {
+        UNSUPPORTED: 0,
+        SUPPORTED: 1,
+        INDETERMINATE: -1
+    };
+
+    /**
+     * Defines optional optimization flags for a {@link Cesium3DTileset}.
+     * Optimizations marked as Cesium3DTileOptimizations.Flags.SUPPORTED or Cesium3DTileOptimizations.Flags.UNSUPPORTED
+     * will not be evaluated at runtime
+     *
+     * @alias Cesium3DTileOptimizations
+     * @constructor
+     *
+     * @example
+     * var tileset = new Cesium.Cesium3DTileset({
+     *     url: ...,
+     *     optimizations: new Cesium.Cesium3DTileOptimizations({
+     *         childrenWithinParent: Cesium.Cesium3DTileOptimizations.Flags.SUPPORTED
+     *     })
+     * });
+     */
+    function Cesium3DTileOptimizations(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * Denotes support for if the childrenWithinParent optimization is supported. This is used to more tightly cull tilesets if children bounds are
+         * fully contained within the parent.
+         *
+         * @type {Cesium3DTileOptimizationFlag}
+         */
+        this.childrenWithinParent = defaultValue(options.childrenWithinParent, Cesium3DTileOptimizations.Flags.INDETERMINATE);
+    }
+
+    var scratchAxis = new Cartesian3();
+
+    /**
+     * Checks and optinonally evaluates support for the childrenWithinParent optimization. This is used to more tightly cull tilesets if children bounds are
+     * fully contained within the parent.
+     *
+     * @param {Cesium3DTile} tile The tile to check.
+     * @param {Boolean} [evaluate=false] Whether to evaluate support if support for the childrenWithinParent optimization is Cesium3DTileOptimizations.INDETERMINATE
+     * @param {Boolean} [force=false] Whether to always evaluate support for the childrenWithinParent optimization
+     * @returns {Boolean} Whether the childrenWithinParent optimization is supported.
+     */
+    Cesium3DTileOptimizations.prototype.checkChildrenWithinParent = function(tile, evaluate, force) {
```

Comment:
I think we can probably just check this every time. In practice, we only need to evaluate it once per tile (unless tiles move relative to their parent?). So really, this is just a small one-time cost on load.